### ISSUE TITLE
[Feature Request] option `-disable-default-keybinginds`

### ISSUE THREAD

**Issue by sunn4room** [gokcehan/lf#1161]
For new comers, default keybindings is great to make `lf` easy to use. But sometimes it's not good:

1. For hackers, it's confused when touch a strange keybinding by mistake, and expect a notification with `unknown mapping: x`.
2. `lf` can act as a *filepicker* for qutebrowser or vivaldi. There is a need to avoid `lf` to do some unrelated things with keybindings.

**Comment by sunn4room** [comment]
Agree.

If I'm right, the root command is `read`, and I can execute all features of `lf` after bind `read`. Maybe it's better to ensure `read` bind, instead of maintain `etc/undefaults`. Anyway. `etc/undefaults` is useful.

**Comment by PowerMan2206** [comment]
@gokcehan Is there an update on this? I'm currently using my own thingy to nuke the default binds, but it's not ideal.

**Comment by gokcehan** [comment]
@PowerMan2206 There are no updates as there are no new suggestions. Do you expect we should maintain `undefaults` in the repo? If you have an undefaults file already, you can submit it as a PR so we might start maintaining in the repository. Why do you say it's not ideal though?

**Comment by PowerMan2206** [comment]
@gokcehan Well, I got it by running `man lf | grep default | grep "'" | cut -d"'" -f2 | | sort | uniq` and editing the output so that it's only keybinds. Then I ran that through `for i in $(cat nuked_defaults); do echo "map '$i'"; done` to put everything in a `map '<bind>'` format, *except* the " and ' characters which I had to put as `map \'` and `map \"`. Then I went through every key and key combination on my keyboard to check if I missed any keys. In other words, it's- uh, it's kinda janky. It works and *probably* takes all the keybinds out of the equasion. I can put it in a PR and send it over for you to test if you wanna.

**Comment by gokcehan** [comment]
@PowerMan2206 I think there are also some additional keybindings which are not listed in your grep results (e.g. `zh`). Also, this would still be missing the command line bindings and default commands. We should ideally have everything altogether if we want to keep this in the repo. In any case, personally I have no interest in this so I probably won't have time to test this myself even if you submit it as a PR.

**Comment by PowerMan2206** [comment]
@gokcehan I took those into account as well ;)

**Comment by joelim-work** [comment]
@gokcehan I think this feature request isn't unreasonable, and that this issue can be reopened.

I was wondering if it was possible to just simply add some kind of `clearmaps` command which deletes all the keybindings but leaves `map ':' read` bound for safety:

```go
case "clearmaps":
	gOpts.keys = make(map[string]expr)
	gOpts.keys[":"] = &callExpr{"read", nil, 1}
```

Below are a few more points that may be worth discussing:

- If the user really wants to unmap `:` too, then they can type `:map ':'` afterwards.
- Even if `:` and `q` are unmapped, it is still possible to quit `lf` by running `lf -remote "send quit"` in another terminal. But this is not ideal because you either need to know the ID of corresponding instance, or have all instances quit.
- It would be possible to add `clearmaps` at the top of the config file so that users can start adding keybindings from a clean state.
- It would also be possible to run `lf -command clearmaps` to 'disable' default keybindings for a single invocation of `lf`.

**Comment by ilyagr** [comment]
This plan sounds reasonable to me (though I'd never use the commands myself).

Optionally, you could also have `clearcmaps` and `clearcmds` for completeness. I don't think `clearcmds` is very important, and I'm not sure whether people would use `clearcmaps`.

For `cmaps`, I think the only key that really needs to be mapped is `<ret>`. Perhaps also `<esc>`, optionally.

I also thought about renaming `clearmaps` to `cleardefaults` and have it nuke all three, but I think that `clearmaps` is more useful if it leaves the cmaps intact.

**Comment by joelim-work** [comment]
Yeah I also considered having this functionality clear literally everything including `cmaps` and `cmds` - I'm not sure what would be an appropriate name though (`cleardefaults` might imply that only the default settings gets cleared, but maybe it's fine). Picking a good name is important because changing it later on would be a breaking change.

As for which `cmaps` to leave (I would leave backspace mapped as well in addition to enter and esc), or even the decision of whether to clear `cmaps` and `cmds` at all, I don't have a strong opinion here and am happy to go with other people's suggestions.

**Comment by ilyagr** [comment]
My (very mild) preference is to have `clearmaps` only clear maps, and (even less importantly) to have the other two commands for completeness.

**Comment by gokcehan** [comment]
@joelim-work I never thought about such commands. They sound quite reasonable.

@ilyagr I don't have any further opinions about the details of such commands, but I think in general I agree with @joelim-work in that the name "default" might imply options in addition to mappings.

**Comment by ilyagr** [comment]
I think we can consider this solved by #1286. If there's any demand for `clearcmaps` or `clearcmds`, we can add those later.
